app/src/main/java/com/amaze/filemanager/adapters/AppsAdapter.java

```diff
@@ -116,22 +121,40 @@ public View getView(final int position, View convertView, ViewGroup parent) {
     View view;
     if (convertView == null) {
       LayoutInflater mInflater =
-          (LayoutInflater) context.getSystemService(Activity.LAYOUT_INFLATER_SERVICE);
+          (LayoutInflater) fragment.getContext().getSystemService(Activity.LAYOUT_INFLATER_SERVICE);
       view = mInflater.inflate(R.layout.rowlayout, null);
       final AppHolder vholder = new AppHolder(view);
       view.findViewById(R.id.generic_icon).setVisibility(View.GONE);
       view.findViewById(R.id.picture_icon).setVisibility(View.GONE);
       view.setTag(vholder);
-      sizeProvider.setView(view.findViewById(R.id.apk_icon));
+      if (sizeProvider != null) {
+        sizeProvider.setView(view.findViewById(R.id.apk_icon));
+      }
     } else {
       view = convertView;
     }
 
     final AppHolder holder = (AppHolder) view.getTag();
 
-    modelProvider.loadApkImage(rowItem.path, holder.apkIcon);
+    if (modelProvider != null) {
+      modelProvider.loadApkImage(rowItem.path, holder.apkIcon);
+    } else if (isBottomSheet) {
+      try {
+        holder.apkIcon.setImageDrawable(
```

Comment:
This seems that it should be in Glide logic, not here.

Reply:
Not using glide for this bottom sheet, just the adapter.

Reply:
Why not? Glide should do most/all image showing. Should this be added as an issue and the PR accepted?

Reply:
I originally used glide but decided to remove it later due ti some reason. That reason is no more applicable, will add it again. Thanks for pointing.

Reply:
https://developer.android.com/kotlin/parcelize